pages/index.vue

```diff
@@ -1,32 +1,90 @@
 <script setup lang="ts">
-interface markerData {
-  latitude: number;
-  longitude: number;
-  name: string;
-  description: string;
-  rating: number;
+import type { MapLocation } from '~/types/mapLocation';
+import type { Coordinate } from '~/types/coordinate';
+
+const mapId = 'mapId';
+const savedLocations = ref<Array<MapLocation>>([]);
+const lastClickedCoordinate = ref<Coordinate | null>(null);
+
+function saveLocation() {
+  if (!lastClickedCoordinate.value) return;
+
+  savedLocations.value.push({
+    coordinates: {
+      latitude: lastClickedCoordinate.value.latitude,
+      longitude: lastClickedCoordinate.value.longitude,
```

Comment:
You could use some destructuring and the object key/value shorthand to avoid some repetition here:

```ts
const { latitude, longitude } = lastClickedCoordinate.value;

saveLocations.value.push({
  coordinates: { latitude, longitude },
  // ...
});
```

Actually I guess this could be even shorted if you're OK to just use `lastClickedCoordinate.value` (assuming the types are happy):
```ts
saveLocations.value.push({
  coordinates: lastClickedCoordinate.value,
  // ...
});
```